crates/iroha_core/src/smartcontracts/wasm.rs

```diff
@@ -749,6 +762,17 @@ impl<W, S> Runtime<state::CommonState<W, S>> {
     }
 }
 
+impl<W, S> Runtime<Option<state::CommonState<W, S>>> {
```

Comment:
The way you have to repeat impls for the `Option<CommonState<...>>` is unfortunate... Would it make sense to make all uses of `CommonState` wrapped into an `Option`? Or would this be too disruptive?